tile/platform/local_machine/mem_chunk.h

```diff
@@ -19,6 +19,9 @@ class MemChunk : public Buffer {
 
   // Gets the chunk's underlying HAL buffer.
   virtual std::shared_ptr<hal::Buffer> hal_buffer() = 0;
+
+  // Decrease uses, return the uses after decrease
+  virtual std::uint64_t DecreaseUses() = 0;
```

Comment:
This is a fragile pattern (it's super easy to get this kind of code wrong, especially when it's subsequently modified by someone else).  Use RAII patterns instead, reifying the notion of "use" as an object that can be managed via `std::shared_ptr`.